### ISSUE TITLE
Support skipping all tests in a suite if skip() is called in setup()

### ISSUE THREAD

**Issue by Arnavion** [theintern/intern#386]
- Add a skip() method to Suite
- If Suite.skip() is called inside Suite.setup(), pre-emptively mark all tests in the suite as skipped without running them.

This would reduce the boilerplate needed if the entire suite needs to be disabled based on some runtime decision:

```javascript
tdd.suite("Check foo", function () {
	var fooSupported = false;

	tdd.before(function () {
		fooSupported = ...;
	});

	tdd.test("Test1", function () {
		if (!fooSupported) {
			this.skip("foo not supported in this environment.");
		}

		...
	});

	tdd.test("Test2", function () {
		if (!fooSupported) {
			this.skip("foo not supported in this environment.");
		}

		...
	});
});
```

into

```javascript
tdd.suite("Check foo", function () {
	tdd.before(function () {
		var fooSupported = ...;
		if (!fooSupported) {
			this.skip("foo not supported in this environment.");
		}
	});

	tdd.test("Test1", function () {
		...
	});

	tdd.test("Test2", function () {
		...
	});
});
```

**Comment by sholladay** [comment]
+1, yes please.

Having to work around this is a common pattern for me.

**Comment by jason0x43** [comment]
Fixed in 4e42a7a3e64580c86c2e5175e15793d50f43da82
